doc/dev/adapter-development-walkthrough.md

```diff
@@ -0,0 +1,824 @@
+# Istio Mixer: Adapter Development Walkthrough
+
+This section walks through step-by-step instructions to implement, test and plug a simple adapter into Mixer. For
+complete details on Adapter life cycle, please refer to the [Adapter Developer Reference Guide](./adapters.md).
+
+**Note**: To complete this walkthrough, it is optional to read the Adapter Developer Reference Guide. However, to
+create a real production quality adapter, it is highly recommended you read the reference guide to better understand
+Adapter lifecycle and various interfaces and objects that Mixer uses to interact with Adapters.
+
+In this walkthrough you're going to create a simple Adapter that:
+
+* Supports the [`metric Template`](../../template/metric/template.proto) which ships with built-in Mixer framework.
+
+* For every request, prints to a file the data it received from Mixer during request time.
+
+**It should approximately take ~30 min to finish this task**
+
+
+# Table of contents
+* [Before you start](#before-you-start)
+* [Step 1: Write basic adapter skeleton code](#step-1-write-basic-adapter-skeleton-code)
+* [Step 2: Write adapter configuration](#step-2-write-adapter-configuration)
+* [Step 3: Link adapter config with adapter code](#step-3-link-adapter-config-with-adapter-code)
+* [Step 4: Write business logic into your adapter](#step-4-write-business-logic-into-your-adapter)
+* [Step 5: Plug adapter into the Mixer](#step-5-plug-adapter-into-the-mixer)
+* [Step 6: Write sample operator config](#step-6-write-sample-operator-config)
+* [Step 7: Start Mixer and validate the adapter](#step-7-start-mixer-and-validate-the-adapter)
+* [Step 8: Write test and validate your adapter (optional)](#step-8-write-test-and-validate-your-adapter-optional)
+* [Step 9: Cleanup](#step-9-cleanup)
+* [Step 10: Next](#step-10-next)
+
+
+# Before you start
+
+Download a local copy of Mixer repo
+
+```
+git clone https://github.com/istio/mixer
+```
+
+Install bazel (version 0.5.2 or higher) from [https://bazel.build/](https://bazel.build/) and add it to your PATH
+
+Set the MIXER_REPO variable to the path where the mixer repository is on the local machine. Example `export MIXER_REPO=$GOPATH/src/istio.io/mixer`
+
+Successfully build Mixer repo.
+
+```
+pushd $MIXER_REPO && bazel build ...
+```
+
+# Step 1: Write basic adapter skeleton code
+
+Create `mysampleadapter` dir and navigate to it.
+
+```
+cd $MIXER_REPO/adapter && mkdir mysampleadapter && cd mysampleadapter
+```
+
+Create file mysampleadapter.go
+
+```
+touch mysampleadapter.go
```

Comment:
Can we get rid of these touch steps? You can just say "create the file mysampleadapter.go with the following content"